src/mongocxx/collection.hpp

```diff
@@ -121,9 +121,9 @@ class MONGOCXX_API collection {
     ///   Optional arguments, see mongocxx::options::aggregate.
     ///
     /// @return A mongocxx::cursor with the results.
-    /// @throws
-    ///   If the operation failed, the returned cursor will throw an exception::query
-    ///   when it is iterated.
+    //
+    /// @throws mongocxx::query_exception when the returned cursor is
```

Comment:
Suggest putting this note in the `@return` section instead of the `@throws` section, since the note is explaining what is being returned by this method, and that this method does not throw.

Relatedly, I suggest adding an `@throws` section to:
- `mongocxx::cursor::begin()`
- `mongocxx::cursor::iterator::iterator()`
- `mongocxx::cursor::iterator::operator++()`
- `mongocxx::cursor::iterator::operator++(int)`